### PR TITLE
Responsive Product Listing Page

### PR DESCRIPTION
**Pull request by Mrudula1205**
## Changes proposed

Designed a product listing page with a filter coloumn that filters the cards according to the category of the product and its colour.Changes were mostly made to the web-version. As I am new to React I couldn't add more features to the page.
Created a nav bar with sign in and cart button (buttons are not responsive)
Added more 3D models. 

## Check List (Check all the applicable boxes) 

- [X ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ X] All new and existing tests passed.

### CODE REVIEW

**Comment by ShwetKhatri2001** [comment]
Good work @Mrudula1205 ! But I'd like you to fix some bugs related to responsiveness I found,

- The container having all elements is not responsive, it's more stretched on large screens and less in size when we apply filter and having less products. The Navbar and Sidebar for filters should not change and all their elements should remain fixed at the top and left respectively. 

image

- The size of the whole container should not be dependant on number of products. It should be taking full width and height always.

image

- Try to make it mobile friendly as well. Fix the sidebar position on the left and show one card in a row for mobile screens.

image

### _You've started well. Let's fix these bugs, improve the feature and get it merged._ 👍

**Comment by Mrudula1205** [comment]
Alright. I have started working on it. Will update soon

**Comment by ShwetKhatri2001** [comment]
Any issues or updates?

**Comment by Mrudula1205** [comment]
I have updated the code and tried fixing the issues you mentioned previously.

File: src/components/ModelViewer/ModelViewer.js
Comment (ShwetKhatri2001) [inline src/components/ModelViewer/ModelViewer.js:1]: Why change `components/ModelViewer` folder and especially `ModelViewer.js`? 
You can create a new folder called `ProductList` and have files `ProductList.js` and `ProductList.css` files there for your changes and can import the ModelViewer there. The App component will be having `ProductList` then.

The main reason is that ModelViewer is going to be reused many times and it should represent a product-card, not the entire list.

**Review by ShwetKhatri2001 — CHANGES_REQUESTED** [review]
You've done a great job, but a few steps away from getting this PR merged! 

- The objects in the top row are getting cut on all screens ( laptop and mobile ).
image

- Resolve the merge conflicts and just rearrange your code a bit as I have mentioned to help other contributors understand the code and contribute to this project.

File: src/components/ModelViewer/ModelViewer.js
Comment (Mrudula1205) [reply src/components/ModelViewer/ModelViewer.js:1]: @ShwetKhatri2001 I have some doubts, and I have reached out to you on Discord regarding the same.

Comment (ShwetKhatri2001) [reply src/components/ModelViewer/ModelViewer.js:1]: I've answered you there.

**Comment by Mrudula1205** [comment]
I have made the necessary changes. CSS was always aligned in my local server, so I wasn't able to check if the changes I made rectified the alignment

**Comment by ShwetKhatri2001** [comment]
> I have made the necessary changes. CSS was always aligned in my local server, so I wasn't able to check if the changes I made rectified the alignment

The issue still persists. Let's focus on resolving this issue and get the PR merged.

First focus on showing the list of cards responsively. 
Remove other things like the Navbar and Sidebar for now. You may add them separately under other issues.

- Sending you this resource to show how it can be implemented easily.
Look at the `Featured Products` and `Recommended Products` at https://salinaka-ecommerce.web.app/

- Check this reusable component `ProductShowcaseGrid` they are using here multiple times to showcase the list of products for different products-data, which will be required in our case as well.
https://github.com/jgudo/ecommerce-react/blob/master/src/components/product/ProductShowcaseGrid.jsx

- In terms of CSS, just one class is responsible for creating this responsive layout - `product-display-grid` and the elements inside the grid take 100% of the provided width.

image

### Hope it helps !

**Comment by Mrudula1205** [comment]
I have updated the code.

**Comment by ShwetKhatri2001** [comment]
Yes, it actually looks good now. But we have new changes specially in `ModelViewer.js` and `App.css`.
The Card is changed as you can see on https://ar-webstore.netlify.app/ as well.

So, you need to pull the latest branch and make your changes working for this card. That's why I told you to have all your changes in a separate component and consider <ModelViewer /> component as a single card in https://github.com/ShwetKhatri2001/AR-Webstore/pull/103#discussion_r1255567400 because it was going to be a card. Although. No worries you can do the same now.

**Comment by ShwetKhatri2001** [comment]
I want to merge this PR now, but you need to update the branch. I can't merge currently since it has conflicting files.

**Comment by Mrudula1205** [comment]
I have resolved the conflicts

**Review by ShwetKhatri2001 — APPROVED** [review]
LGTM, Merging it. Truly appreciate your efforts 👍

Congrats on your first PR getting merged on AR-Webstore @Mrudula1205 🎊 Keep Contributing.

**Pull request merged**